OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -85,10 +85,16 @@ public override bool Tick(Actor self)
 					return false;
 
 				case PickupState.LockCarryable:
-					if (!carryable.LockForPickup(cargo, self))
+					var lockResponse = carryable.LockForPickup(cargo, self);
+					if (lockResponse == LockResponse.Failed)
 						Cancel(self);
 
-					state = PickupState.Pickup;
+					if (lockResponse == LockResponse.Success)
+						state = PickupState.Pickup;
+
+					if (lockResponse == LockResponse.Pending)
+						QueueChild(new FlyIdle(self, 1));
```

Comment:
This makes the TS carryall stop 4 cells away (see the `Intercept` case above) while the unit sorts itself out.  We should keep flying towards the target and trigger the idle case only when we reach it.

Reply:
One option would be to set `ChildHasPriority = false` and replace the state machine with a single queued `Land` right from the start. We can then do a range check before ticking the child and start calling `LockForPickup` every tick once we're closer than four cells. If the carryall gets within say 1 cell (these two ranges should ideally be exposed to yaml) we can cancel the land, queue a FlyIdle, then another land.

Reply:
We might consider moving this logic inside the `Land` activity since this behaviour is applicable whenever an aircraft tries to dock with a mobile unit. We could rework `LockForPickup` into an `ILockForLanding` interface that can be implemented by `Carryable`, but also by, for example, a future `Carrier` trait to implement RA2 style aircraft carriers and destroyers.

Reply:
That's probably making life more difficult than it needs to be right now - what if we kept it in `PickupUnit` for now, which limits the scope of testing and regressions, then we look to move it into `Land` as part of the docking overhaul in Next + N?

Reply:
Done.